pkg/exchange/coinbase/exchage.go

```diff
@@ -308,25 +310,36 @@ func (e *Exchange) QueryKLines(ctx context.Context, symbol string, interval type
 		log.Warnf("limit %d is greater than the maximum limit 300, set to 300", options.Limit)
 		options.Limit = DefaultKLineLimit
 	}
-	granity := interval.String()
+	granity := fmt.Sprintf("%d", interval.Seconds())
 	req := e.client.NewGetCandlesRequest().ProductID(toLocalSymbol(symbol)).Granularity(granity)
 	if options.StartTime != nil {
 		req.Start(*options.StartTime)
 	}
 	if options.EndTime != nil {
 		req.End(*options.EndTime)
 	}
-	rawCandles, err := req.Do(ctx)
+	res, err := req.Do(ctx)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to get klines(%v): %v", interval, symbol)
 	}
-	if len(rawCandles) > options.Limit {
-		rawCandles = rawCandles[:options.Limit]
-	}
-	klines := make([]types.KLine, 0, len(rawCandles))
-	for _, rawCandle := range rawCandles {
-		candle := rawCandle.Candle()
-		klines = append(klines, toGlobalKline(symbol, granity, candle))
+	candles := make([]api.Candle, 0, len(res))
+	for _, c := range res {
+		candles = append(candles, *c.Candle())
+	}
```

Comment:
Most of us are accustomed to using pointers, for example make([]*api.Candle, 0, len(res)), so you don't have to dereference.

```
candles := make([]*api.Candle, 0, len(res))
	for _, c := range res {
		candles = append(candles, c.Candle())
	}
```

Reply:
As your review for PR #1909, I will apply the error handling here for invalid raw candles.
In that case, I think we don't need to use pointers.